### ISSUE TITLE
Cannot deploy Next.js app with Server Side Rendering

### ISSUE THREAD

**Issue by dominoFire** [firebase/firebase-tools#7146]
### [REQUIRED] Environment info

**firebase-tools:**

```
PS C:\Users\ferag\one\two\three\four> firebase --version
13.8.1
```

**Platform:** Windows 11, node --version: v20.10.0

### [REQUIRED] Test case

This happened when I was trying to deploy a next.js app, noted in https://github.com/copilotkit/copilotkit

### [REQUIRED] Steps to reproduce

Create a next app with `create-next-app`

```shell
npx --yes create-next-app@14.2.3
```

Add a next.js backend route. Example: src/app/route.ts, like so:

```typescript
export async function POST(req: Request): Promise<Response> {
    return new Response('Hello World', { status: 200 });
}
```

Init firebase Hosting in the created app

```
firebase init
```

Deploy the project

```
firebase deploy --debug
```

### [REQUIRED] Expected behavior

My app is deployed to Firebase

### [REQUIRED] Actual behavior

The deploy fails with the following message:

```

Building a Cloud Function to run this application. This is needed due to:
 • Image Optimization
 • non-static component /api/route

[2024-05-09T18:25:24.695Z] >>> [apiv2][query] GET https://firebasehosting.googleapis.com/v1beta1/projects/propio-panel/sites/propio-panel [none]
[2024-05-09T18:25:25.324Z] <<< [apiv2][status] GET https://firebasehosting.googleapis.com/v1beta1/projects/propio-panel/sites/propio-panel 200
[2024-05-09T18:25:25.325Z] <<< [apiv2][body] GET https://firebasehosting.googleapis.com/v1beta1/projects/propio-panel/sites/propio-panel {"name":"projects/propio-panel/sites/propio-panel","defaultUrl":"https://propio-panel.web.app","type":"DEFAULT_SITE"}
[2024-05-09T18:25:25.427Z] Error: ENOENT: no such file or directory, lstat 'C:\Users\ferag\one\two\three\four\hosting\.next\C:\Users\ferag\one\two\three\four\hosting\.next\app-build-manifest.json'

Error: An unexpected error has occurred.
```

**Comment by dominoFire** [comment]
I have seen this issue reported before: https://github.com/firebase/firebase-tools/issues/6253

**Comment by dominoFire** [comment]
Also, seen here https://github.com/firebase/firebase-tools/issues/5158

**Comment by dominoFire** [comment]
Seen similar issue here: https://github.com/firebase/firebase-tools/issues/6435

**Comment by dominoFire** [comment]
Did some debugging. This seems happening because glob function is returning absolute paths instead of relative paths that are needed later in a copying process:


https://github.com/firebase/firebase-tools/blob/e06ae7b1b00a3f131803c20b57337cdc510cea9f/src/frameworks/next/utils.ts#L475-L487

**Comment by dominoFire** [comment]
I can attempt a PR.

**Comment by aalej** [comment]
Hey @dominoFire, thanks for reporting this. I was able to reproduce the error you encountered. I’ll raise this to our engineering team so they can take a look. To workaround the issue, could you try deploying using firebase-tools v13.8.0? 

It also looks like you may have identified the cause of the issue here, feel free to create a PR for this if you’re up for it! Contributions are always appreciated!

**Comment by jamesdaniels** [comment]
@leoortizz PTAL too since you've been fighting some path fires

**Comment by dominoFire** [comment]
Just noticed the glob package version used here is 7.*, new [new glob version](https://www.npmjs.com/package/glob) is 10.*, and it behaves differently with realpath parameter.

On 7.* (in a test app) it return the full path on Windows
On 10.* (in a test app) it returns the relative path


https://github.com/firebase/firebase-tools/blob/143ffd014c249cf0223c93e71c1c4b3aeb972265/package.json#L124

**Comment by leoortizz** [comment]
thanks @dominoFire for reporting the issue and apologies for the inconvenience. I have a fix in #7148 and explained the reason in the PR description. As soon as that PR is merged and released this issue will be fixed.

As a workaround, please use version `v13.6.1` in the meantime.
